### PR TITLE
feat/EditGroupViewModel

### PR DESCRIPTION
**Pull request by Atilla13**
### Summary

This PR implements the Edit Group functionality, allowing users to modify existing group details including name, category, and description. The implementation follows clean architecture principles with comprehensive validation and error handling.

### What's Changed

**New Files**
- EditGroupViewModel.kt - ViewModel managing edit group state and business logic
- EditGroupViewModelTest.kt - Comprehensive test suite with +70 test cases

**Implementation Details**
EditGroupViewModel
- Architecture: Follows MVVM pattern with Repository abstraction, never directly accessing Firebase/Firestore
- State Management: Uses StateFlow for reactive UI updates with well-defined state machine
- Core Functionality:
1. loadGroup(groupId) - Fetches existing group data from repository
2. updateGroup(groupId) - Saves modified group data while preserving other properties (members, events, owner)
3. Real-time validation for all fields with immediate user feedback
4. Proper loading and error states throughout the flow

**Validation Rules**
Name: Required, 3-30 characters, letters/numbers/spaces/underscores only, no consecutive spaces
Description: Optional, max 300 characters
Category: Required, one of Social/Activity/Sports

**State Machine**
Loading → Idle/Editing → Submitting → Success/Error
Proper loading indicators during async operations
Clear error messages for all failure scenarios
Success state with group ID for navigation

**Testing**
EditGroupViewModelTest - test cases organized into categories:
- [x]  Initial state verification
- [x] Group loading with success/error paths
- [x] Name validation edge cases
- [x] Description validation
- [x] Category management
- [x] Form validity logic
- [x] Update success scenarios
- [x] Update error handling
- [x] Integration tests
- [x] State management
Test Coverage: All public methods, validation rules, error paths, and edge cases

### Technical Notes

- Uses coroutines with proper viewModelScope management
- Trims input for validation but preserves original for display
- Preserves immutable group properties (ownerId, memberIds, eventIds) during updates
- Exception handling with user-friendly error messages
- No direct Firebase dependencies - fully testable with mocked repository

### Checklist

- Follows Repository Pattern and MVVM architecture
- Comprehensive validation with clear error messages
- Proper state management with loading/error/success states
- Complete test coverage
- Error handling for all edge cases
- KDoc documentation for all public APIs

### CODE REVIEW

File: app/src/main/java/com/android/joinme/ui/groups/EditGroupViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+          trimmedName.contains(Regex("\\s{2,}")) -> "Multiple consecutive spaces not allowed"
+          !trimmedName.matches(Regex("^[a-zA-Z0-9_ ]+$")) ->
+              "Only letters, numbers, spaces, and underscores allowed"
+          else -> null
+        }
+    _uiState.value =
+        _uiState.value.copy(
+            name = name,
+            nameError = error,
+            isValid = computeValidity(name, error, _uiState.value.descriptionError))
```
Comment (vlonfat) [inline app/src/main/java/com/android/joinme/ui/groups/EditGroupViewModel.kt]: You assign name instead of trimmedName, please fix it
